source/base/partitioner.cc

```diff
@@ -277,7 +278,8 @@ namespace Utilities
               import_targets_temp.push_back(std::pair<unsigned int,
                                             unsigned int> (i, receive_buffer[i]));
             }
-        import_targets_data = import_targets_temp;
+
+        swap(import_targets_temp, import_targets_data);
```

Comment:
I agree. If we do this, though, then we should wait until the C++11 support requirement is merged.